### DISCUSSION TITLE
Access pods inside colima kubernetes cluster with curl

### DISCUSSION THREAD

**Discussion by gotpist1** [abiosoft/colima#570] [Q&A]
Hello! We have switched from minikube and using Colima with kubernetes. Everything fires up and internaly, pods can talk to eachother. Now, i want to be able to curl a pods health endpoint but with no prevail. I have started an ingress-controller following:

https://github.com/abiosoft/colima/issues/133

And that seems to work well. But still, i cant call a pod from my host machine. Any insight on what might be missing?

**Comment by abiosoft** [comment]
@gotpist1 are the pods exposed behind a service? If yes, LoadBalancer or NodePort?

**Reply by gotpist1** [reply]
It's ClusterIp services. But i have enabled an nginx ingress controller inside the cluster. Then i have a config map that explains the different adressess.

**Reply by gotpist1** [reply]
Im thinking that i might need to expose the controller to my host with a NodePort service maybe.

**Reply by gotpist1** [reply]
Ok. So i followed this approach instead: https://blog.dustinrue.com/2022/08/colima-kubernetes-and-helm-chart-testing/comment-page-1/#comment-61503

But that doesn't work either. Everything seems to be in order, but still gets connection refused when trying to curl my host. FUDGE!

**Comment by abiosoft** [comment]
Can you share your setup or yaml files? Or maybe run something simple at first.

What is the output of `kubectl get services`?

**Reply by gotpist1** [reply]
The output of `kubectl get services --all-namespaces` is 

![Screenshot 2023-01-07 at 12 20 23](https://user-images.githubusercontent.com/15625831/211147599-b05606a2-037c-411c-a91c-edd69fa7ccfc.png)

**Reply by gotpist1** [reply]
If i run a curl against the external ip with https i get this;

curl https://192.168.106.2/userpod/monitor/info/
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

So, my guess is that the loadbalancer is infact working. But it cant resolve the host i have stated

**Reply by gotpist1** [reply]
And `kubectl get ingress` gives me this:

![Screenshot 2023-01-07 at 12 24 22](https://user-images.githubusercontent.com/15625831/211147841-a2d1132a-03c2-4642-ac98-395c72742016.png)

**Reply by abiosoft** [reply]
Is the ingress not reachable without https? i.e. http://192.168.106.2/userpod/monitor/info/

**Reply by gotpist1** [reply]
Also, if i run curl against the external ip without https, i hit the nginx but get a 404 not found. That is to be expected, because in my ingress i specify the host seen above. 

I have added a entry in my hosts file that points the external ip against my host aswell.

**Reply by gotpist1** [reply]
I need to be able to resolve the host, so i can use tls and let the ingress route me to the right place.

This might be an issue with my companys setup, but it worked fine on intel mac with minikube :(

**Reply by gotpist1** [reply]
if i use curl with the --resolve flag, it works. But that is not really ideal.

curl --resolve 'dmdp.localhost:443:192.168.106.2' https://dmdp.localhost/userpod/monitor/info -k

**Reply by gotpist1** [reply]
Just wanted to let you know that i solved this. Apparently, using an alias that ends with ".localhost" made everything act up real bad. As soon as i switched to ".local" everything worked as expected :)
